contrib/swank-asdf.lisp

```diff
@@ -387,7 +387,11 @@ Example:
 AND in its source-registry. (legacy name)"
   (unique-string-list
    (mapcar
-    #'pathname-name
+    (lambda (asd)
+      (typecase asd
+        (pathname (pathname-name asd))
+        (string asd)
+        (t asd)))
```

Comment:
instead of this typecase, could we perhaps consistently produce strings from the collection below?

Reply:
How about this, producing pathnames? https://github.com/slime/slime/commit/2fdbad58ac801f98d4f7e4aa3bf1304e26633ce1

Reply:
How about using `:collect (lambda (path) (c (pathname-name path)))`?  Does that work?

Reply:
It would also work
But it looks simpler with collecting strings, take a look at the last commit